tests/php/src/Factories/OrderFactory.php

```diff
@@ -0,0 +1,225 @@
+            ),
+            array(
+                'product_id' => $this->factory->product->create( array( 'name' => 'Test Product 2' ) ),
+                'quantity'   => 1,
+            ),
+        );
+    }
+
+    /**
+     * Add a discount or coupon line item to the Order.
+     *
+     * @param array $order_data
+     * @param array $products
+     * @param array $coupon_data
+     * @return WC_Order
+     */
+    public function add_discount( \WC_Order $order, \WC_Coupon $coupon ) {
+        $order->apply_coupon( $coupon->get_code() );
+
+        // Calculate totals
+        $order->calculate_totals();
+
+        // Save the order
+        $order->save();
+
+        return $order;
+    }
+
+    public function set_item_coupon( WC_Coupon $coupon ) {
+        $this->coupon_items[] = $coupon;
+
+        return $this;
+    }
+
+    public function set_item_fee( array $args = [
+		'name' => 'Extra Fee',
+		'amount' => 100,
+	] ) {
+        $args = wp_parse_args(
+            $args, [
+				'name' => 'Extra Fee',
+				'amount' => 100,
+			]
+        );
+
+        $fee_item = new WC_Order_Item_Fee();
+        $fee_item->set_name( $args['name'] );
+        $fee_item->set_total( $args['amount'] );
+
+        $this->order_items[] = $fee_item;
+
+        return $this;
+    }
+
+    public function set_item_shipping( array $args = [] ) {
+        $args = wp_parse_args(
+            $args, [
+				'name' => 'Flat shipping',
+				'amount' => 25,
+			]
+        );
+
+        $shipping_item = new WC_Order_Item_Shipping();
+        $shipping_item->set_name( $args['name'] );
+        $shipping_item->set_total( $args['amount'] );
+
+        $this->order_items[] = $shipping_item;
+
+        return $this;
+    }
+
+    /**** Wrapper of WC Test */
+
+    /**
+     * Delete an order.
+     *
+     * @param int $order_id ID of the order to delete.
+     */
+    public static function delete_order( $order_id ) {
+        return WC_Helper_Order::delete_order( $order_id );
+    }
+
+    /**
+     * Create a order.
+     *
+     * @since   2.4
+     * @version 3.0 New parameter $product.
+     *
+     * @param int        $customer_id The ID of the customer the order is for.
+     * @param WC_Product $product The product to add to the order.
+     * @param int        $product_quantity The product quanitty to add to the order.
+     *
+     * @return WC_Order
+     */
+    public static function create_order( $customer_id = 1, $product = null, $product_quantity = 1 ) {
+        return WC_Helper_Order::create_order( $customer_id, $product, $product_quantity );
+    }
+
+    /**
+     * Helper function to create order with fees and shipping objects.
+     *
+     * @param int        $customer_id The ID of the customer the order is for.
+     * @param WC_Product $product The product to add to the order.
+     *
+     * @return WC_Order
+     */
+    public static function create_order_with_fees_and_shipping( $customer_id = 1, $product = null ) {
```

Comment:
We should use the factory to assign the default customer ID.

Reply:
This is method signature of `WC_Helper_Order` so that we can replace the current  uses of `WC_Helper_Order` through `OrderFactory` though regex where method signature should be same.